### PR TITLE
Topic4

### PR DESCRIPTION
**Pull request by matihaure**
:+1:

### CODE REVIEW

**Comment by amilcar-infante** [comment]
Matias, please could you remove the items related to topic 1 that shouldn't be in this topic to help me to do a cleaner code review? Or better, could you use a topic4 branch please?
Thanks in advance

**Comment by matihaure** [comment]
Im sorry , I cant at the moment , im having some issues with my internet. I
will try to fix it tomorrow.

2015-02-26 0:14 GMT-03:00 amilcar-infante notifications@github.com:

**Comment by amilcar-infante** [comment]
Sure no problem

File: Topic4Services/src/cartServices/Counter.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package cartServices;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Counter {
+
+	private static Counter Counter = null;
+	private static final AtomicInteger count = new AtomicInteger(0);
```
Comment (amilcar-infante) [inline Topic4Services/src/cartServices/Counter.java:8]: Nice!!!

File: Topic4Services/src/cartServices/ShoppingCartImpTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+public class ShoppingCartImpTest {
+
+	private Cart cart;
+	private Payment paymentMethod;
+	private List<Item> items;
+	private ShoppingCartService shoppingCartService;
+	private Container container;
+
+	@Before
```
Comment (amilcar-infante) [inline Topic4Services/src/cartServices/ShoppingCartImpTest.java:25]: Excellent!

File: Topic4Services/src/cartServices/ShoppingCartImpTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	@Test
+	public void getItemsTest() {
+
+		assertEquals(container.items(), shoppingCartService.getItems());
+
+	}
+
+}
```
Comment (amilcar-infante) [inline Topic4Services/src/cartServices/ShoppingCartImpTest.java:89]: You should create tests with different types of payment

**Comment by amilcar-infante** [comment]
This pull request will be close since Matias created a new pull request named Topic4-Updated to have a cleaner review

**Pull request closed without merging**